iocore/net/UnixNet.cc

```diff
@@ -525,7 +531,17 @@ NetHandler::mainNetEvent(int event, Event *e)
       close_UnixNetVConnection(vc, trigger_event->ethread);
     else if (vc->read.enabled && vc->read.triggered)
       vc->net_read_io(this, trigger_event->ethread);
-    else if (!vc->read.enabled) {
+    else if (vc->read.error) {
+      int err = 0, errlen = sizeof(int);
+      if (getsockopt(vc->con.fd, SOL_SOCKET, SO_ERROR, &err, (socklen_t *)&errlen) == -1) {
+        err = errno;
+      }
+      if (err != EAGAIN && err != EINTR) {
+        vc->readSignalError(this, err);
```

Comment:
Sorry, my mistaken here. We should get vio mutex locked first before callback to SM. the new suggest :
```
else if (vc->read.enabled && vc->read.triggered || vc->read.error)
```
move "else if (vc->read.error) { ... }" into "vc->net_read_io()"

we can not access any member of NetVC ( e.g. vc->con.fd ) before get vio mutex locked.